Configure

```diff
@@ -1670,6 +1670,7 @@ while (<PIPE>) {
   $predefined{$1} = defined($2) ? $2 : "";
 }
 close(PIPE);
+my $old_xcode_uses_makedepend = defined($predefined{__APPLE_CC__}) && !defined($predefined{__clang__});
```

Comment:
As first reaction I wanted to suggest to assign more generic name to this variable, like no_cc_as_makedepend, but then I figured that it would probably be more appropriate to consolidate the decision logic. Something like this perhaps:
```
my $makedepprog;
if ($predefined{__GNUC__} >= 3
   && !($predefined{__APPLE_CC__}) && !$predefined{__clang__})) {
    $makedepprog = $cc;
}
...
    s/^MAKEDEPPROG=.*$/MAKEDEPPROG = $makedepprog/ if defined($makedepprog);
```
One can also resolve that cross-compile prefix thing in the first if...

Reply:
Yeah, I wasn't sure if I should make it a bit more generic, or keep it specific to OS X.

With respect to the cross-compile prefix, do you mean that `$randlib` and `$ar`, etc are missing in the variable substitutions?

I wasn't planning on fixing the cross-compile section, as it might actually have a cc that supports -M...

Reply:
I didn't mean that you would revamp *that* if, only make cross-compile assignment to MAKEDEPEND in *this* if. Which would allow to move MAKEDEPPROG substitution *alone* out *that* if. But this is optional! If in doubt, just make the variable more generic and consolidate logic. Just in case, rationale behind assigning generic name is "what if there is another special case like this".

Reply:
The cross-compile section still looks wrong, however; it's missing the `$ar` and `$randlib`, etc variables. But, that's out of scope here.